schematic/schemas/json_schema_generator.py

```diff
@@ -0,0 +1,747 @@
+        name: What to name the object
+        is_required: Whether or not the property is required by the schema
+        description: a description of the property
+
+    Returns:
+        JSON object
+    """
+    schema: dict[str, Any] = {name: {"description": description}}
+    one_of_list: list[dict[str, Any]] = [{"enum": sorted(enum_list), "title": "enum"}]
+    if not is_required:
+        one_of_list += [{"type": "null", "title": "null"}]
+    schema[name]["oneOf"] = one_of_list
+    return schema
+
+
+def _create_simple_property(
+    name: str, property_data: PropertyData, is_required: bool, description: str
+) -> dict[str, Any]:
+    """
+    Creates a JSON Schema property
+    If a property_type is given the type is added to the schema
+    If a property_type is not given and is_required is  "not: {type:null} is added
+
+    Arguments:
+        name: What to name the object
+        property_data: Info parsed about the property from the validation rules
+        is_required: Whether or not the property is required
+        description: a description of the property
+
+    Returns:
+        JSON object
+    """
+    schema: dict[str, Any] = {name: {"description": description}}
+
+    if property_data.property_type and is_required:
+        schema[name]["type"] = property_data.property_type
+    elif property_data.property_type:
+        schema[name]["oneOf"] = [
+            {"type": property_data.property_type, "title": property_data.property_type},
+            {"type": "null", "title": "null"},
+        ]
+    elif is_required:
+        schema[name]["not"] = {"type": "null"}
+
+    if property_data.range_min is not None:
+        schema[name]["minimum"] = property_data.range_min
+    if property_data.range_max is not None:
+        schema[name]["maximum"] = property_data.range_max
+
+    return schema
+
+
+def _get_property_data_from_validation_rules(
+    validation_rules: list[str],
+) -> PropertyData:
+    """
+    Returns data from validation rules in the form of a PropertyData dataclass
+
+    Args:
+        validation_rules: A list of validation rules
+
+    Returns:
+        PropertyData: validation rule data
+    """
+    property_type: Union[str, None] = None
+    is_array = False
+    range_min: Union[float, None] = None
+    range_max: Union[float, None] = None
+
+    if validation_rules:
+        if rule_in_rule_list("list", validation_rules):
+            is_array = True
+
+        type_rule = _get_type_rule_from_rule_list(validation_rules)
+        if type_rule:
+            property_type = TYPE_RULES.get(type_rule)
+
+        range_rule = _get_in_range_rule_from_rule_list(validation_rules)
+        if range_rule:
+            if property_type not in ["number", "integer"]:
+                property_type = "number"
+            range_min, range_max = _get_ranges_from_range_rule(range_rule)
+
+    return PropertyData(property_type, is_array, range_min, range_max)
```

Comment:
It may make more sense to add `validation_rules` as an attribute on `PropertyData` and accomplish this logic within a `__post__init__` function